src/posix/platform/netif.cpp

```diff
@@ -860,13 +860,14 @@ static void logAddrEvent(bool isAdd, bool isUnicast, struct sockaddr_in6 &addr6,
     OT_UNUSED_VARIABLE(addr6);
     OT_UNUSED_VARIABLE(addressString);
 
-    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY)) ||
+    if ((error == OT_ERROR_NONE) || ((isAdd) && (error == OT_ERROR_ALREADY || error == OT_ERROR_INVALID_ARGS)) ||
```

Comment:
Could you add comment to explain why `OT_ERROR_INVALID_ARGS` is identical to `OT_ERROR_ALREADY`?

Reply:
It's  not identical, but equally harmless. 

`OT_ERROR_INVALID_ARGS` is returned in two cases:
1. the multicast address is invalid (i.e. not starting with `ff`)
2. posix-nets adds a multicast address like ff02::2 to Thread stack, which is considered a `fixed` multicast address.

Case 1 is apparently wrong, but case 2 is fine. What really happens in practice is case 2, so ignoring `OT_ERROR_INVALID_ARGS` should be fine in most cases.

Maybe we can use another error code (e.g. `OT_ERROR_REJECTED` ) for case 2 so that it can also be ignored gracefully.

Reply:
@wgtdkp suggested to not call `otIp6SubscribeMulticastAddress` for fixed addresses like `ff02::2`, which would effectively require us to add a new API like `otIp6IsFixedMulticastAddress` so that posix-netif can use to check if a multicast address is fixed.

Maybe returning `OT_ERROR_REJECTED` can be a simpler solution which effectively gives platform-netif enough information it needs. 
Moreover, returning `OT_ERROR_INVALID_ARGS` for case 2 is not suitable because `ff02::2` is a valid multicast address.

Reply:
FYI: reverted back to original commit without this chagne.